Lib/fontTools/t1Lib/__init__.py

```diff
@@ -179,43 +178,41 @@ def readLWFN(path, onlyHeader=False):
 
 def readPFB(path, onlyHeader=False):
 	"""reads a PFB font file, returns raw data"""
-	f = open(path, "rb")
-	data = []
-	while True:
-		if f.read(1) != bytechr(128):
-			raise T1Error('corrupt PFB file')
-		code = byteord(f.read(1))
-		if code in [1, 2]:
-			chunklen = stringToLong(f.read(4))
-			chunk = f.read(chunklen)
-			assert len(chunk) == chunklen
-			data.append(chunk)
-		elif code == 3:
-			break
-		else:
-			raise T1Error('bad chunk code: ' + repr(code))
-		if onlyHeader:
-			break
-	f.close()
-	data = bytesjoin(data)
-	assertType1(data)
-	return data
+	with open(path, "rb") as f:
+		data = []
+		while True:
+			if f.read(1) != bytechr(128):
+				raise T1Error('corrupt PFB file')
+			code = byteord(f.read(1))
+			if code in [1, 2]:
+				chunklen = stringToLong(f.read(4))
+				chunk = f.read(chunklen)
+				assert len(chunk) == chunklen
+				data.append(chunk)
+			elif code == 3:
+				break
+			else:
+				raise T1Error('bad chunk code: ' + repr(code))
+			if onlyHeader:
+				break
+		data = bytesjoin(data)
+		assertType1(data)
+		return data
 
 def readOther(path):
 	"""reads any (font) file, returns raw data"""
-	f = open(path, "rb")
-	data = f.read()
-	f.close()
-	assertType1(data)
+	with open(path, "rb") as f:
+		data = f.read()
+		assertType1(data)
```

Comment:
this line could be de-dented, doesn't need to be inside the with block

Reply:
actually the whole block after this line need not be indented under `with`.
in general I like to release the file resource as soon as possible and avoid too long indented blocks